core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -0,0 +1,413 @@
+      language_code;
+    this.userService.getUserInfoAsync().then(userInfo => {
+      this.username = userInfo.getUsername();
+      this.isCurriculumAdmin = userInfo.isCurriculumAdmin();
+    });
+    this.userService.getUserContributionRightsDataAsync().then(
+      userContributionRights => {
+        this.userCanReviewTranslationSuggestionsInLanguages = (
+          userContributionRights
+            .can_review_translation_for_language_codes);
+        this.canEditTranslation = (
+          this.userCanReviewTranslationSuggestionsInLanguages.includes(
+            this.languageCode) && this.username !== this.activeSuggestion.
+            author_name
+        );
+      });
+    this.errorMessage = '';
+    this.errorFound = false;
+    this.startedEditing = false;
+    this.resolvingSuggestion = false;
+    this.lastSuggestionToReview = this.remainingContributions.length <= 0;
+    this.translationHtml = (
+      this.activeSuggestion.change.translation_html);
+    this.status = this.activeSuggestion.status;
+    this.contentHtml = (
+      this.activeSuggestion.change.content_html);
+    this.explorationContentHtml = (
+      this.activeSuggestion.exploration_content_html);
+    this.isHtmlContent = (
+      this.activeSuggestion.change.data_format === 'html'
+    );
+    this.isUnicodeContent = (
+      this.activeSuggestion.change.data_format === 'unicode'
+    );
+    this.isSetOfStringsContent = (
+      this.activeSuggestion.change.data_format ===
+        'set_of_normalized_string' ||
+      this.activeSuggestion.change.data_format ===
+        'set_of_unicode_string'
+    );
+    this.reviewMessage = '';
+    if (!this.reviewable) {
+      this.suggestionIsRejected = (
+        this.activeSuggestion.status === 'rejected');
+      if (this.suggestionIsRejected) {
+        this._getThreadMessagesAsync(this.activeSuggestionId);
+      }
+    }
+  }
+
+  updateSuggestion(): void {
+    const updatedTranslation = this.editedContent.html;
+    const suggestionId = this.activeSuggestion.suggestion_id;
+    this.preEditTranslationHtml = this.translationHtml;
+    this.translationHtml = updatedTranslation;
+    this.contributionAndReviewService.updateTranslationSuggestionAsync(
+      suggestionId,
+      updatedTranslation,
+      () => {
+        this.translationUpdated = true;
+        this.startedEditing = false;
+        this.contributionOpportunitiesService.
+          reloadOpportunitiesEventEmitter.emit();
+      },
+      this.showTranslationSuggestionUpdateError);
+  }
+
+  // The length of the commit message should not exceed 375 characters,
+  // since this is the maximum allowed commit message size.
+  generateCommitMessage(): string {
+    var contentId = this.activeSuggestion.change.content_id;
+    var stateName = this.activeSuggestion.change.state_name;
+    var contentType = contentId.split('_')[0];
+
+    var commitMessage = `${contentType} section of "${stateName}" card`;
```

Comment:
New lines above and below seem unnecessary?

Reply:
Done, by mistake